src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -172,6 +175,18 @@ public override void VisitProperty(IPropertySymbol symbol)
             }
         }
 
+        private static bool ShouldBeNullableAnnotated(ISymbol symbol, FlowAnalysisAnnotations annotations)
+        {
+            if (symbol is Symbols.PublicModel.Symbol csharpSymbol)
```

Comment:
>if (symbol is Symbols.PublicModel.Symbol csharpSymbol) [](start = 12, length = 54)

Attributes is not C# specific concept, symbol model allows to check for them even when we are dealing with non-C# symbols. I guess it is fine to use this as an optimization, but it feels wrong to limit the behavior just to C# symbols.

Reply:
I rely on the `GetFlowAnalysisAnnotations` extension method which is defined on `Microsoft.CodeAnalysis.CSharp.Symbol`. So this approach relies on CSharp symbols. As I said [earlier](https://github.com/dotnet/roslyn/pull/50527/files#r558384945), I'm not sure about doing the right thing at the right place at all:

Why is `((IPropertySymbol)symbol).Type.NullableAnnotation` not `NullableAnnotation.Annotated` in this case?

Reply:
> Why is ((IPropertySymbol)symbol).Type.NullableAnnotation not NullableAnnotation.Annotated in this case? 

I am not following. The type isn't annotated. Right?

---
In reply to: [558496330](https://github.com/dotnet/roslyn/pull/50527#discussion_r558496330) [](ancestors = 558496330)

Reply:
No. the type is not annotated. Another solution would have been to make sure it is annotated and I wasn't sure what the right thing to do is here: Tackle the problem in SymbolVisitor or make sure the type is annotated.

Reply:
>  Another solution would have been to make sure it is annotated and I wasn't sure what the right thing to do is here: Tackle the problem in SymbolVisitor or make sure the type is annotated.

I don't think we would want to change behavior of the ```Type``` on the symbol itself. The annotation exposed there is about an annotation from source, i.e. about the presence or lack of the `?` in source and how the fact is interpreted.

---
In reply to: [558553322](https://github.com/dotnet/roslyn/pull/50527#discussion_r558553322) [](ancestors = 558553322)